ColesStopAndShop/CashRegister.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            CreditBalance = creditBalance;
+            DebitBalance = debitBalance;
+            StoreId = storeId;
+        }
+
+        /// <summary>
+        /// Processes payment with given fact that it is debit/cash or credit.
+        /// </summary>
+        /// <param name="isDebitOrCash">The fact that it is debit/cash or credit.</param>
+        public void ProcessPayment(bool isDebitOrCash)
+        {
+            decimal totalCostOfPurchase = 0;
+            List<int> listOfItemPricesAtPurchase = new List<int>();
+            List<ItemId> listOfItems = new List<ItemId>();
+
+            // Scans each item given to clerk. 
+            do
+            {
+                //Console.Clear();
+                Console.WriteLine("Key (9) to process payment.");
+
+                Console.WriteLine("Enter item id: ");
+                int.TryParse(Console.ReadLine(), out int itemScanned);
+
+                if (itemScanned == 9)
+                {
+                    break;
+                }
+
+                if (!Enum.IsDefined(typeof(ItemId), itemScanned))
+                {
+                    Console.WriteLine("Invalid item id.");
+                    Console.WriteLine("Press any key to retry...");
+                    Console.ReadKey();
+                    continue;
+                }
+                else
+                {
+                    listOfItems.Add((ItemId)itemScanned);
+
+                    Console.WriteLine("Enter cost of item: ");
+                    int priceOfItem = int.Parse(Console.ReadLine());
+
+                    listOfItemPricesAtPurchase.Add(priceOfItem);
+                }
+
+            } while (true);
+
+            // After items are scanned; affect till balance and get total cost.
+            foreach(int itemPrice in listOfItemPricesAtPurchase)
+            {
+                totalCostOfPurchase += itemPrice;
+
+                if (!isDebitOrCash)
+                {
+                    CreditBalance += itemPrice;
+                }
+                else
+                {
+                    DebitBalance += itemPrice;
+                }
+            }
+
+            totalCostOfPurchase *= salesTaxRate;
+
+            // Converts items to string for receipt.
+            List<string> itemsBoughtToString = new List<string>();
+            foreach (ItemId item in listOfItems)
+            {
+                itemsBoughtToString.Add(item.ToString());
+            }
+
+            /*
+             * Prints Receipt below 
+             */
+
+            Console.WriteLine($"Thanks for stopin' and shopin' at Cole's Stop and Shop!");
+
+            Console.WriteLine("Items: ");
+
+            for (int i = 0; i < listOfItems.Count; i++)
+            {
+                for (int j = 0; j < listOfItems.Count; j++)
+                {
+                    Console.WriteLine($" {listOfItemPricesAtPurchase[i]} :  {itemsBoughtToString[i]}");
+                }
+            }

```

Comment:
- [ ] Refactor to a single loop. The 2D loop here is not required and is causing #27.